be/src/exec/pipeline/pipeline_driver_queue.h

```diff
@@ -30,37 +46,66 @@ class DriverQueue {
     virtual void put_back_from_dispatcher(const std::vector<DriverRawPtr>& drivers) = 0;
 
     virtual StatusOr<DriverRawPtr> take() = 0;
+    virtual StatusOr<DriverRawPtr> take(size_t* queue_index) = 0;
```

Comment:
Maybe we needn't this interface, and make `QuerySharedDriverQueue::put_back()` record `queue_index` in `PipelineDriver::_dispatch_queue_index` like `DriverQueueWithouLock::put_back()`.